### PR TITLE
Fix SSR useId memory leak

### PR DESCRIPTION
**Pull request by snowystinger**
Closes https://github.com/adobe/react-spectrum/issues/5260

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

**Review by reidbarber — APPROVED** [review]

File: packages/@react-aria/utils/src/useId.ts [hunk 1]
```diff
@@ -31,7 +31,9 @@ export function useId(defaultId?: string): string {
     nextId.current = val;
   }, []);
 
-  idsUpdaterMap.set(res, updateValue);
+  if (typeof window !== 'undefined') {
```
Comment (yarastqt) [inline packages/@react-aria/utils/src/useId.ts:34]: Unfortunately Deno have window object at server context, maybe use `useIsSSR` or `canUseDom`?

Comment (snowystinger) [reply packages/@react-aria/utils/src/useId.ts:34]: thanks for the reminder about Deno

File: packages/@react-aria/ssr/src/SSRProvider.tsx [hunk 2]
```diff
@@ -93,7 +93,7 @@ export function SSRProvider(props: SSRProviderProps): JSX.Element {
   return <LegacySSRProvider {...props} />;
 }
 
-let canUseDOM = Boolean(
+export let canUseDOM = Boolean(
```
Comment (snowystinger) [inline packages/@react-aria/ssr/src/SSRProvider.tsx:96]: unfortunately, useIsSSR doesn't work in all cases, I'm not sure why.
this check works for our use case here, if we don't want to export it, i could copy it over, it's quite small

**Review by reidbarber — APPROVED** [review]
LGTM, I'm fine exporting this

**Review by iamwillpowell — APPROVED** [review]

File: packages/@react-aria/ssr/src/SSRProvider.tsx (hunk 2 above)
Comment (devongovett) [reply packages/@react-aria/ssr/src/SSRProvider.tsx:96]: ya copy it

**Comment by tadatuta** [comment]
Thank you very much for such fast fix! Can you please release new version with it?

@snowystinger @iamwillpowell

**Comment by snowystinger** [comment]
It will be in our next release which should be coming up soon. In the meantime, this is a very small fix, you could easily cover it with patch-package while waiting for our next release.

**Pull request merged**